lighthouse-core/lib/tracehouse/main-thread-tasks.js

```diff
@@ -262,10 +262,36 @@ class MainThreadTasks {
             // It's ending at traceEndTs, it means we were missing the end event. We'll truncate it to the parent.
             nextTask.endTime = currentTask.endTime;
             nextTask.duration = nextTask.endTime - nextTask.startTime;
+          } else if (
+            nextTask.startTime - currentTask.startTime < 1000 &&
+            !currentTask.children.length
+          ) {
+            // The parent started less than 1ms before the child, we'll let it slide by swapping the two,
```

Comment:
can you add a graphic like this in here?

```
    ████████████currentTask.parent██████████████████
            █████████nextTask██████████████
           ███████currentTask███████
```

its so great

Reply:
oooooo, I love it ❤️